From 3d0cdd8af1b415712eeb00e377c307001684ad06 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Wed, 21 Jun 2017 15:35:20 +0300
Subject: [PATCH] Move releasever check after the etc/dnf/vars substitutions.

The substitutions may actually set the releasever correctly,
and so the check is premature.

Upstream-Status: Submitted [https://github.com/rpm-software-management/dnf/pull/868]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 dnf/cli/cli.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dnf/cli/cli.py b/dnf/cli/cli.py
index c53c2a52..f9f2c13a 100644
--- a/dnf/cli/cli.py
+++ b/dnf/cli/cli.py
@@ -882,12 +882,12 @@ class Cli(object):
             releasever = dnf.rpm.detect_releasever(conf.installroot)
         elif releasever == '/':
             releasever = dnf.rpm.detect_releasever(releasever)
-        if releasever is None:
-            logger.warning(_("Unable to detect release version (use '--releasever' to specify "
-                             "release version)"))
         conf.releasever = releasever
         subst = conf.substitutions
         subst.update_from_etc(conf.installroot)
+        if releasever is None:
+            logger.warning(_("Unable to detect release version (use '--releasever' to specify "
+                             "release version)"))
 
         for opt in ('cachedir', 'logdir', 'persistdir'):
             conf.prepend_installroot(opt)
-- 
2.11.0

